modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDmlTest.java

```diff
@@ -672,6 +676,39 @@ public void testDeleteUsingCompositePk() {
         assertQuery("SELECT d FROM test").returnNothing().check();
     }
 
+    @Test
+    public void testOperationFailsToCommitWhenDetectsSchemaChange() throws InterruptedException {
+        sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
+
+        CountDownLatch startDml = new CountDownLatch(1);
+        CountDownLatch latch = new CountDownLatch(1);
```

Comment:
can be replaced with barrier.
```suggestion
        CyclicBarrier barrier = new CyclicBarrier(2);
```